### DISCUSSION TITLE
Cargo Check is run on every keyinput

### DISCUSSION THREAD

**Discussion by jpx40** [zed-industries/zed#19793] [Feature Requests]
Didn't wanted to add an issue for this yet i just wanted to say that Cargo Check is run on every keyinput which slows down zed extreme. it would be better when it some delay.

**Comment by SomeoneToIgnore** [comment]
rust-analyzer has "cargo check on save" feature enabled by default, so right after we update the file contents, r-a reruns the checks.
Zed has `"autosave": "off",` by default, so I'm not really sure what happens on your side.

It would be interesting to know whether you have a specific configuration issue or Zed had gone rogue and saves the file on every keystroke.

Other autosaving options include the delay one, as a workaround:

https://github.com/zed-industries/zed/blob/2e32f1c8a19525004cb9b3b7ac9cc53c9aa4fdd0/assets/settings/default.json#L473-L484

**Reply by jpx40** [reply]
i will try saving with delay, i have autosave on. I just noticed today that it kinda gone huhi.

**Comment by jpx40** [comment]
can i disable cargo check in Zed i mean ist clippy + rust analyzer enough. For me it would be ok running it only on compiles.

**Reply by SomeoneToIgnore** [reply]
If you can read the docs, you can do it for sure.
https://zed.dev/docs/languages/rust#more-server-configuration

What I am interested in though, is 

> Cargo Check is run on every keyinput

It should not run on key inputs, but on save.
Is it a bad description? Is it really running on keypress?

**Reply by jpx40** [reply]
i don't know how i can describe it. rust-anaylyzer kinda had some latency. Disabling cargo check fixed it.

**Comment by jpx40** [comment]
i think latenzy kinda was an rust-ana issue. i hangs sometimes on large builds cargo clean fixes it.
